data/campaigns/The_Rise_Of_Wesnoth/scenarios/17c_Troll_Hole.cfg

```diff
@@ -136,14 +131,31 @@
         {LOYAL_UNIT 4 (Tentacle of the Deep) 11 15}
         {LOYAL_UNIT 5 (Tentacle of the Deep) 19 17}
 #endif
-#ifdef HARD
 
+#ifdef HARD
         {LOYAL_UNIT 4 (Tentacle of the Deep) 11 15}
         {LOYAL_UNIT 5 (Tentacle of the Deep) 19 17}
         {LOYAL_UNIT 2 (Tentacle of the Deep) 11 17}
         {LOYAL_UNIT 3 (Tentacle of the Deep) 19 15}
 #endif
 
+        # Both tags need a "+" here in order to work:
+#define SPIDER_TRAIT
+        [+unit]
+            [+modifications]
+#ifdef EASY
+                {TRAIT_AGED}
+#endif
+#ifdef NORMAL
+                {TRAIT_SLOW}
+#endif
+#ifdef HARD
+                {TRAIT_RESILIENT}
+#endif
+            [/modifications]
+        [/unit]
+#enddef
+
         {RANDOM 1..4}
```

Comment:
Mainly to @CelticMinstrel and @mattsc: I'm wondering if it would make sense to have a standard method to choose X unique options from a list.

```
# result is a comma-separated list in $random
[set_variable]
    rand=1..4
    count=2
[set_variable]
```

Or have a `rand` option for `[set_variables]`

Reply:
I think this already works:

```
[set_variable]
    rand=list,of,things
    name=choice
[/set_variable]
```

Adding a rand option in `[set_variables]` seems like a good idea too; however, `[set_variables]` is implemented in C++ so it's more work to extend it.

Reply:
@CelticMinstrel my feature selection was to be able to get more than 1 option from the list.

The block of code that I'm commenting on wants to choose 2 of the 4 corners of the map, but instead has to map each of 1, 2, 3 or 4 to a pair of 2 corners.

Reply:
Ahh, I see. Adding a count option in `[set_variable]rand=` would be fairly easy in principle, but what would be the output format?

Reply:
Good point @CelticMinstrel . I was thinking it would be a comma-separated list, but that isn't a good format for driving a `[for]` or `[foreach]` loop.

Reply:
Right, if it was a comma-separated list, you'd probably need to pass it thru `[set_variables]split=` anyway… so maybe it's better to put this in `[set_variables]` in the first place?